automates/text_reading/src/main/scala/org/clulab/aske/automates/OdinActions.scala

```diff
@@ -817,9 +834,201 @@ class OdinActions(val taxonomy: Taxonomy, expansionHandler: Option[ExpansionHand
+          if (arg._2.exists(_.label == "Identifier")) {
+            identifierInputs ++= arg._2.filter(_.label.contains("Identifier"))
+            phraseInputs ++= arg._2.filterNot(_.label.contains("Identifier"))
+          } else phraseInputs ++= arg._2
+          // if there is an identifier input, check if there's an overlap between the identifier input and other phrase inputs
+          if (identifierInputs.nonEmpty) {
+            for (i <- identifierInputs) {
+              val inputNumCheck = new ArrayBuffer[Mention]
+              if (phraseInputs.nonEmpty) {
+                for (p <- phraseInputs) {
+                  newInputs.append(p)
+                  val overlappingInterval = i.tokenInterval.overlaps(p.tokenInterval)
+                  // if there's no overlap, append the identifier input to the inputNumCheck
+                  if (overlappingInterval == false) {
+                    inputNumCheck.append(i)
+                  }
+                }
+                // if the number of identifier inputs appended to the inputNumCheck is the same as the number of phrase inputs,
+                // it means that there is no overlap, so attach the identifier input to newInputs.
+                // if the number is not the same, it means there is an overlap, so don't attach.
+                if (inputNumCheck.length == phraseInputs.length) newInputs.append(i)
+                // if there's no phrase inputs, just append the identifier inputs to the newInputs.
+              } else newInputs.append(i)
+            }
+            // if there's no identifier inputs, just append the phrase inputs to the newInputs.
+          } else newInputs ++= phraseInputs
+        } else outputs ++= arg._2
+      }
+      // make new arguments with newInputs and outputArgs
+        val newArgs = Map("input" -> newInputs.distinct, "output" -> outputs)
+        val newFunctions = copyWithArgs(m, newArgs)
+        toReturn.append(newFunctions)
+    }
+    toReturn
+  }
+
+  def filterOutputOverlaps(mentions: Seq[Mention], state: State): Seq[Mention] = {
+    val phraseTokInt = new ArrayBuffer[Interval]
+    val newMentions = new ArrayBuffer[Mention]
+    val groupMens = mentions.groupBy(m => (m.sentence, m.asInstanceOf[EventMention].trigger.tokenInterval, m.foundBy))
+    for (group <- groupMens) {
+      if (group._2.head.arguments("output").nonEmpty) {
+        val (identOutputMen, phraseOutputMen) = group._2.partition(_.arguments("output").head.label.contains("Identifier"))
+        if (identOutputMen.nonEmpty) {
+          for (i <- identOutputMen) {
+            val outputNumCheck = new ArrayBuffer[Mention]
+            if (phraseOutputMen.nonEmpty) {
+              for (p <- phraseOutputMen) {
+              val overlappingInterval = i.arguments("output").head.tokenInterval.overlaps(p.arguments("output").head.tokenInterval)
+              if (overlappingInterval == false) {outputNumCheck.append(i)}
```

Comment:
@alicekwak please make the same change here as I requested in my comment above on line 922.

Reply:
Yes I will, thanks!